### ISSUE TITLE
Sensor Value Drops

### ISSUE THREAD

**Issue by MarcoNica** [a529987659852/openwbmqtt#42]
I use the openwb in combination with a SolarEdge inverter. 
Ever since I integrated the entities for purchased electricity, grid feed-in and total photovoltaic yield into the HomeAssistant energy dashboard, I have noticed that they sometimes return a zero value and shortly afterwards return to the correct value.
This messes up the energy dashboard, as it always adds the jump from 0 to the old value to its sum.
![image](https://user-images.githubusercontent.com/17566860/212002309-da128aa4-21d9-4d7e-9e1c-8ff7b9d588aa.png)

Total Photovoltaic Yield value jumps to a lower value for a period of time, but not to 0. Of course, this also messes up the energy dashboard.
![image](https://user-images.githubusercontent.com/17566860/212002351-828916cb-7f81-4ba2-95ec-b9f5cae2f753.png)

**Comment by ChristophCaina** [comment]
Hi,
I had similar issues for a long time...
Usually, these "drops" can happen when the MQTT Broker persists messages and due to a restart of either the wallbox or HomeAssistant, it will send the older persistent / saved values.

To prevent that, I've removed the MQTT bridge and connected HomeAssistant directly to the MQTT Broker of the Wallbox.
Also, to not rely on the PV Data comming from the Wallbox, I've added a Modbus Proxy into my network, which is getting all data from the Inverter and sending it to the wallbox and homeassistant solaredge Modbus integration.

Solar Edge Modbus Integration:
https://github.com/binsentsu/home-assistant-solaredge-modbus

Modbus Proxy (HA Addon):
https://github.com/Akulatraxas/ha-modbusproxy

Modbus Proxy (Standalone application):
https://github.com/tiagocoutinho/modbus-proxy

**Comment by MarcoNica** [comment]
@ChristophCaina thanks for your quick reply.
My current setup is configured like descriped in the readme of this openWB integration an rely on Mosquitto broker.

Could you tell me the steps i need to do?
How to connect HomeAssistant directly to the MQTT Broker of the wallbox?

**Comment by ChristophCaina** [comment]
it's pretty straight forward.
With the Standard configuration you are using an MQTT Server running in HomeAssistant.
This MQTT Server will then be configured to connect to the MQTT Broker running on the OpenWB.

Your OpenWB Integration will then be configured to connect to the Broker running in HomeAssistant.

I've just configured the OpenWB Integration to connect to the IP from the OpenWB wallbox directly, without having any other MQTT Broker.

**Comment by alxw1** [comment]
Good Morning @ChristophCaina, i am following your discussion because i am trying to set up q bridge between openwb and homeassistant following all those guides all over the internet and tbh it is very frustrating since no data from openwb shows up in homeassistant (and also not as a knot in homeassistant opening the mqtt in Mqtt-Explorer)

therefore would you be so Kind to provide a little guide how you direct connected the openwb in homeassistant mqtt?

**Comment by MarcoNica** [comment]
@ChristophCaina 
If i get you right i firts can clear my setup from anything related to the openwb integration. So remove Mosquitto and the openwb integration completly from my current setup.


How to configure the MQTT Server in HomeAssistant?
And how to configure tje OpenWB to connect to the IP of the Wallbox?
As @alxw1 mentioned it would be wonderfull if you can make a little tutorial.

**Comment by ChristophCaina** [comment]
Of course, the above ist only working If you don't use anything else that requires a mqtt Broker (zigbee2mqtt for example)

**Comment by MarcoNica** [comment]
@ChristophCaina 
Ok, thanks for that tutorial.
But i need the normal mqtt Broker as you mentioned above.


So my plan now is to get my needed solaredge data directly from modbus and not from the openwb.
Can you please tell me what to setup in the openwb?

**Comment by alxw1** [comment]
> If you have set up a Username and Password for MQTT in your OpenWB, you need to add this to the configuration.

Thanks a lot. Just to be Safe… did you Set up a new mqtt in your openwb? Because i only have the cloud one and that one needs a User + PW.  but opening the openwb in mqtt-explorer no user and pw are needed. 

This is how it is der up:
![856EC385-49C8-442C-92F7-952F4BE4ECDC](https://user-images.githubusercontent.com/122490647/212044577-d08fd4df-93c2-41f7-ac12-535ab64b5cc8.jpeg)

**Comment by MarcoNica** [comment]
@alxw1 I think you can completly disable this. So set "MQTT-Brücke" to "Aus". The MQTT Broker(not Brige) keeps active and as you mentioned is reachable in the local network without any username and password.

**Comment by alxw1** [comment]
It drives me nuts…

If i put in the OpenWBs IP-Adress and leave User and PW blank. The HA cant Connect. 

On mqtt Explorer I van Connect and receive values. 

![Uploading 7CA4B305-255C-4EAB-B4F3-33A6ABA698D9.jpeg…]()

**Comment by MarcoNica** [comment]
@alxw1 
In your screenshot you have to add the ip adress of your open wb.

**Comment by alxw1** [comment]
Thats What i did:

here you can See my openwb is Sending 
![1B220D82-4A3C-4088-BE34-7ED851A51707](https://user-images.githubusercontent.com/122490647/212050266-317b6a9b-ac4f-410a-850f-0e1723766464.png)

And the openwbs ip Adresse
![DA2D1175-7638-4A6F-8C2E-F43EB142FFA2](https://user-images.githubusercontent.com/122490647/212050425-b876f423-1bf1-4d8b-aff1-cc571803d867.jpeg)

putting the *.50 in here 
![7FA04BEC-9D4F-4FB9-8170-81F357837CA9](https://user-images.githubusercontent.com/122490647/212050543-4aa2f04a-8def-40c2-a3c2-0f0682b88c3f.jpeg)

Ends up like this:

![0F426536-1FFB-4AE7-AFD4-CB7AC764FC31](https://user-images.githubusercontent.com/122490647/212050723-a39c7e74-0657-4546-83a3-b76d9ec85606.jpeg)

**Comment by pcmaka** [comment]
I have the same issue and changed 
```
cleansession true
```
to
```
cleansession false
```

in my bridge config and this seems to have result the issues for me.

Per mosquitto documentation, this prevents retained messages to be send again on reconnect.
